core/src/main/java/org/ehcache/core/spi/cache/AbstractValueHolder.java

```diff
@@ -181,6 +181,6 @@ public long getId() {
 
   @Override
   public String toString() {
-    return "" + value();
+    return "" + invoke();
```

Comment:
I am not going down the rabbit hole of making the `AbstractValueHolder` aware of values being materialised or not.